lib/fs.ts

```diff
@@ -164,7 +164,7 @@ class FSPath {
 			return;
 		}
 
-		this.writeUpdateNow(dataFetcher, options);
+		return this.writeUpdateNow(dataFetcher, options);
```

Comment:
This current return value is meaningless which is why it's not propagated.

I think if we wanted to return a Promise from `writeUpdate`, it should be a meaningful Promise.

Reply:
A meaningful response might be a Promise that resolves the next time the update was written. This would require the `if (pendingUpdate)` branch to return a Promise.

Reply:
I'm not sure how it's meaningless? It returns `writeUpdateNow` which returns `safeWrite` who only resolves when the file is written to.